token/program-2022/src/processor.rs

```diff
@@ -7309,6 +7316,54 @@ mod tests {
         assert_eq!(account_account, account3_account);
     }
 
+    #[test]
+    fn initialize_account_on_non_transferable_mint(){
+        let program_id = crate::id();
+        let account = Pubkey::new_unique();
+        let mut account_account = SolanaAccount::new(
+            account_minimum_balance(),
+            Account::get_packed_len(),
+            &program_id,
+        );
+        let owner_key = Pubkey::new_unique();
+        let mut owner_account = SolanaAccount::default();
+        let mint_key = Pubkey::new_unique();
+        let mint_len =
+            ExtensionType::try_calculate_account_len::<Mint>(&[ExtensionType::NonTransferable])
+                .unwrap();
+        let mut mint_account = SolanaAccount::new(Rent::default().minimum_balance(mint_len),
+                               mint_len,
+                               &program_id);
+        let mut rent_sysvar = rent_sysvar();
+
+        // create a non-transferable mint
+        assert_eq!(
+            Ok(()),
+            do_process_instruction(initialize_non_transferable_mint(&program_id, &mint_key).unwrap(),
+            vec![&mut mint_account, &mut rent_sysvar],
```

Comment:
The rent sysvar isn't required for this instruction, it only needs the mint
```suggestion
            vec![&mut mint_account],
```

Reply:
removed

Reply:
Thank you for the suggestions.
as for the lint check, after I ran the ./cargo-nightly clippy -Zunstable-options --all-targets command, It showed  2 errors that refer to a commented-out code, I supposed this would not impact the  CI.

Reply:
Seem like I broke more than one test, Will take a look later :)

Reply:
Ah yes, a few tests depend on the old behavior unfortunately, so these are legitimate test failures. Let me know if you're having a hard time with these and I can help out

Reply:
I've fixed all the failed tests in the token-2022, could u help take a look over the changes and check whether it looks good, thank you